backends/p4tools/common/CMakeLists.txt

```diff
@@ -5,16 +5,24 @@ include(common)
 
 project(p4tools-common)
 
-# Store the current git commit hash as part of version information.
-execute_process(
-  COMMAND git log --pretty=format:-%h -n 1
-  OUTPUT_VARIABLE P4TOOLS_VERSION_HASH
-  ERROR_QUIET
+# Pull in a specific version of Z3 and link against it.
+set(P4TOOLS_Z3_VERSION "4.11.2")
+message("Fetching Z3 version ${P4TOOLS_Z3_VERSION} for P4Tools...")
+
+# Print out download state while setting up Z3.
+set(FETCHCONTENT_QUIET_PREV ${FETCHCONTENT_QUIET})
+set(FETCHCONTENT_QUIET OFF)
+fetchcontent_declare(
+  z3
+  URL https://github.com/Z3Prover/z3/releases/download/z3-${P4TOOLS_Z3_VERSION}/z3-${P4TOOLS_Z3_VERSION}-x64-glibc-2.31.zip
```

Comment:
Oh, I have not noticed before that you are using a binary release of Z3, not building from source. In that case, I strongly suggest to use it only as default to simplify build on Linux + x64, but allow the user to override with their own Z3. The reason is this is actually quite fragile:

- The build uses GLIBC 2.31, while this is OK for Ubuntu 20/Debian 10 but it will not run on RHEL 8 that has 2.28 (even though `p4c` will build there if newer GCC is used).
- It is build with GCC 9.4 (`objdump -s --section .comment libz3.so`), while we support GCC >= 9.1 (which is probably compatible), the problem is this will break builds with Clang + libc++ -- while not usual on Linux, someone could be using this.
- It only works on Linux + x86. I presume `p4c` can be also build on macOS and it is quite likely it would build on *BSD, but neither of these will work with this Z3 library.

---

Therefore what I suggest is to allow the user to specify `Z3_INCLUDE_DIR` and `Z3_LIBRARY` the way they could before. If these are not specified, than some rudimentary check should be done and than Z3 fetched:
- check that [`CMAKE_SYSTEM_NAME`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SYSTEM_NAME.html#cmake-system-name) is `Linux`
- check that [`CMAKE_SYSTEM_PROCESSOR`](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html) is `x86_64`.

Without setting the paths, the install should proceed as you suggested here. This way most users (who use Linux with usual setup) will benefit from this while the other will not need to modify CMakeLists or do some hack to override Z3.